src/mlpack/core/data/imputer.hpp

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file imputer.hpp
+ * @author Keon Kim
+ *
+ * Defines Imputer(), a utility function to replace missing variables
+ * in a dataset.
+ */
+#ifndef MLPACK_CORE_DATA_IMPUTER_HPP
+#define MLPACK_CORE_DATA_IMPUTER_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * This class implements a way to replace target values. It is dependent on the
+ * user defined StrategyType and MapperType used to hold dataset's information.
+ *
+ * @tparam Option of imputation strategy.
+ * @tparam MapperType that is used to hold dataset information.
+ * @tparam primitive type of input and output's armadillo matrix.
+ */
+template<typename T, typename MapperType, typename StrategyType>
+class Imputer
+{
+ public:
```

Comment:
Could we offer another api, let the users store the StrategyType too?Maybe in some cases, users need to setup some parameters of the StrategyType